Chore: migrate repack native modules to turbo modules (#344)
callstack/repack: README.md

- [x] Code signing and verification
- [x] `@callstack/repack-init` - tool that automates the integration of the Re.Pack into React-Native projects

### Planned features

Available at: [Projects](https://github.com/callstack/repack/projects?type=classic)

- [ ] `ChunksToHermesBytecodePlugin` plugin to automatically transform async chunks to bytecode format.
- [ ] Code signing and verification

> - [ ] `webpack-init` command

Reviewer: This whole section is irrelevant – these features are already implemented & released. Either this branch needs sync with `main,` or conflicts were resolved in a wrong way

- [ ] Code signing and verification
- [ ] `webpack-init` command

> ### New Architecture Support (Turbo Module)

Reviewer 2: ```suggestion
### New Architecture Support
```

Author: Will fix. Thanks.

----

> This library supports new architecture!. [turbo module](https://reactnative.dev/docs/next/the-new-architecture/pillars-turbomodules) offers a new architecture for initializing native modules.

Reviewer 3: ```suggestion
This library supports new architecture! Using [turbo modules](https://reactnative.dev/docs/next/the-new-architecture/pillars-turbomodules) offers a new way of initializing native modules.
```
